internal/validations/validations.go

```diff
@@ -78,11 +85,16 @@ func validateGoSymbols(_ context.Context, path string, baton *Baton) *types.Vali
 
 	if goLessThan118.Check(baton.GoVersion) {
 		return nil
+	} else if goGreaterThanOrEqualTo122.Check(baton.GoVersion) {
+		if !golang.ExpectedSyms(requiredGolangSymbolsPost122, symtable) {
+			return types.NewValidationError(types.ErrGoMissingSymbols)
+		}
+	} else {
+		if !golang.ExpectedSyms(requiredGolangSymbolsPre122, symtable) {
+			return types.NewValidationError(types.ErrGoMissingSymbols)
+		}
```

Comment:
Some logic is duplicated here (and if we are to make another change if the future, it will be worse).

What about:

```go
requiredGolangSymbols := requiredGolangSymbolsPre122
if goGreaterThanOrEqualTo122.Check(baton.GoVersion) {
	requiredGolangSymbols = requiredGolangSymbolsPost122
}
```